packages/@core.triggers/index.js

```diff
@@ -0,0 +1,107 @@
+/*
+* Triggers manager.
+* - add triggers
+* - drop triggers
+* - execute triggers validation triggers conditions using *json-rules-engine* and similar rules language
+*
+* */
+
+const Q = require('q')
+const { Engine, Rule } = require('json-rules-engine')
+const get = require('lodash/get')
+
+class TriggersManager {
+    constructor (triggersMap) {
+        this.engine = new Engine()
+        this.triggers = triggersMap || new Map()
+    }
+
+    addTrigger (trigger) {
+        const { rule, action } = trigger
+        const triggerName = get(rule, ['event', 'type'])
+
+        if (!triggerName) {
+            throw new Error('Trigger event type was not provided')
+        }
+
+        if (this.triggers.has(triggerName)) {
+            throw new Error(`Trigger ${triggerName} is already registered. Rule event type should be unique`)
+        }
+
+        const engineRule = new Rule(rule)
+
+        this.triggers.set(triggerName, { rule: engineRule, action })
+        this.engine.addRule(engineRule)
+    }
+
+    dropTrigger (trigger) {
+        const { rule } = trigger
+        const triggerName = get(rule, ['event', 'type'])
+
+        if (!this.triggers.has(triggerName)) {
+            throw new Error(`There is no Trigger with name: ${triggerName} found, unable to drop`)
+        }
+
+        this.triggers.delete(triggerName)
+        this.engine.removeRule(rule)
+    }
+
+    flushTriggers () {
+        this.triggers.forEach(( triggerName, { rule }) => {
+            this.engine.removeRule(rule)
+        })
+
+        this.triggers.clear()
+    }
+
+    executeTrigger (data, context) {
+        if (this.triggers.size > 0) {
+            return this.engine.run(data)
+                .then(({ events }) => {
+                    const actions = events.map((event) => {
+                        const trigger = this.triggers.get(event.type)
+
+                        if (!trigger) {
+                            throw new Error(`Trigger with name ${event.params.name} is not found`)
+                        }
+
+                        const { action } = trigger
+
+                        if (!action) {
+                            throw new Error(`Trigger action from ${event.params.name} is not found`)
+                        }
```

Comment:
Why not to use `Ajv` to validate JSON schema?

Reply:
Which schema u writing about?)

Reply:
I have highlighted validations of a rule schema, wrote procedurally.
It seems, like it will be simpler to have declarative description of JSON-schemas in `domains/ticket/triggers/*`.

Reply:
I'm not sure we need JSON schema validator here. json-rules-engine got nested validators, u can test it by yourself.